lib/model/yake_keywords.py

```diff
@@ -7,8 +7,26 @@
 from lib import schemas
 
 import yake
+from langdetect import detect
 
 class Model(Model):
+
+    def keep_largest_overlapped_keywords(self, keywords):
```

Comment:
An easier to read implementation here would be to first sort keywords by length. Then the loop can only compare each keyword to the keywords that are the same or longer length.

This function should have unit tests. I don't think we'd see this situation, but if we had the same keyword appear twice in a list (e.g., `['Apple', 'Apple', 'Banana']`), I think we would not include `Apple` at all in the output as this is currently written (a unit test would confirm this). We should include one instance, however.

Reply:
I am not sure if the suggested implementation will cover all the cases.

I've added unit tests for this 
```
        keywords_test = [('Alegre',0), ('AlegreAlegre', 0), ('Alegre Alegre', 0), ("Presto", 0)]
        expected = [('AlegreAlegre', 0), ('Alegre Alegre', 0), ('Presto', 0)]
```

Reply:
Added more test cases 
        keywords_test = [('Alegre', 0),('Alegre', 0),('Timpani', 0), ('Presto Timpani', 0), ('AlegreAlegre', 0), ('Alegre Alegre', 0), ("Presto", 0)]
        expected = [('Presto Timpani', 0), ('AlegreAlegre', 0), ('Alegre Alegre', 0)]